File: frontend/main-app/app/src/main/java/com/sh22/frontend_networking_test/MainActivity.java
package com.sh22.frontend_networking_test;

import android.os.Bundle;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentTransaction;


import com.sh22.frontend_networking_test.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {

    ActivityMainBinding binding;
    Float score = 0.7f;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        binding = ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());
        replaceFragment(HomeFragment.newInstance(score.toString()));

        // Navigation bar stuff
        binding.bottomNavigationView.setOnNavigationItemSelectedListener(item ->{
            switch(item.getItemId()){
                case R.id.home:
                    replaceFragment(HomeFragment.newInstance(score.toString()));
                    break;
                case R.id.appliances:
                    replaceFragment(new AppliancesFragment());
                    break;
                case R.id.ecosystem:
                    replaceFragment(EcosystemFragment.newInstance(score.toString()));
                    break;
                case R.id.tips:
                    replaceFragment(new TipsFragment());
                    break;
                case R.id.settings:
                    replaceFragment(new SettingsFragment());
                    break;
            }

            return true;
        });
    }

    private void replaceFragment(Fragment fragment){
        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.frame_layout, fragment);
        fragmentTransaction.commit();
    }


}

File: frontend/main-app/app/src/main/java/com/sh22/frontend_networking_test/MainActivity.kt
package com.sh22.frontend_networking_test

import android.app.PendingIntent.getActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import androidx.appcompat.app.AppCompatActivity
import androidx.cardview.widget.CardView
import com.unity3d.player.UnityPlayer


class MainActivity : AppCompatActivity() {
    var mUnityPlayer: UnityPlayer? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Set a onClick callback for the test button
        val button = findViewById<Button>(R.id.test_button)
        button.setOnClickListener(View.OnClickListener {
            UnityPlayer.UnitySendMessage("HealthManager", "SetHealth", "1.0")
        });
    }

    override fun onDestroy() {
        mUnityPlayer?.quit()
        super.onDestroy()
    }

    override fun onPause() {
        super.onPause()
        mUnityPlayer?.pause()
    }

    override fun onResume() {
        super.onResume()
        mUnityPlayer?.resume()
    }
}